src/Cli/dotnet/commands/InstallingWorkloadCommand.cs

```diff
@@ -221,6 +221,18 @@ protected void UpdateWorkloadManifests(WorkloadHistoryRecorder recorder, ITransa
                 if (updateToLatestWorkloadSet)
                 {
                     resolvedWorkloadSetVersion = _workloadManifestUpdater.GetAdvertisedWorkloadSetVersion();
+                    var currentWorkloadVersionInfo = _workloadResolver.GetWorkloadVersion();
+                    if (resolvedWorkloadSetVersion != null && currentWorkloadVersionInfo.IsInstalled && !currentWorkloadVersionInfo.WorkloadSetsEnabledWithoutWorkloadSet)
+                    {
+                        var currentPackageVersion = WorkloadSetVersion.ToWorkloadSetPackageVersion(currentWorkloadVersionInfo.Version, out var currentWorkloadSetSdkFeatureBand);
+                        var advertisedPackageVersion = WorkloadSetVersion.ToWorkloadSetPackageVersion(resolvedWorkloadSetVersion, out var advertisedWorkloadSetSdkFeatureBand);
+
+                        if (currentWorkloadSetSdkFeatureBand > advertisedWorkloadSetSdkFeatureBand ||
+                            new NuGetVersion(currentPackageVersion) >= new NuGetVersion(advertisedPackageVersion))
```

Comment:
I believe the answer is 'yes', but the package version is guaranteed to be <= 3 parts since it omits the '0', and that's why you needed to switch to comparing package versions, whereas the real version may be 4 parts, correct?

Reply:
Yes, basically we switch to the (semantic) NuGet package version here because we have version comparison implemented for that, but not for the potentially 4-part with preview suffix workload set version.